staging/src/k8s.io/client-go/tools/cache/reflector_test.go

```diff
@@ -574,23 +574,27 @@ func TestReflectorFullListIfExpired(t *testing.T) {
 			}
 			switch options.ResourceVersion {
 			case "0":
-				return &v1.PodList{ListMeta: metav1.ListMeta{ResourceVersion: "10"}, Items: pods[0:4]}, nil
+				if options.Continue == "" && options.Limit == 4 {
+					return &v1.PodList{ListMeta: metav1.ListMeta{ResourceVersion: "10"}, Items: pods[0:4]}, nil
+				}
+				t.Fatalf("Unrecognized Continue (%s) or Limit (%v)", options.Continue, options.Limit)
 			case "10":
+				if options.Continue != "" {
+					t.Fatalf("specifying resource version (%s) is not allowed when using continue (%s).", options.ResourceVersion, options.Continue)
+				}
 				switch options.Limit {
 				case 4:
-					switch options.Continue {
-					case "":
-						return &v1.PodList{ListMeta: metav1.ListMeta{Continue: "C1", ResourceVersion: "11"}, Items: pods[0:4]}, nil
-					case "C1":
-						return nil, apierrs.NewResourceExpired("The resourceVersion for the provided watch is too old.")
-					default:
-						t.Fatalf("Unrecognized Continue: %s", options.Continue)
-					}
-				case 0:
+					return &v1.PodList{ListMeta: metav1.ListMeta{Continue: "C1", ResourceVersion: "11"}, Items: pods[0:4]}, nil
```

Comment:
leave the `switch options.Continue` in place with the "" and default cases, and just drop the "C1" case

Reply:
would this be easier to follow if we fully specify every accepted rv/continue/limit combination that results in a return, rather than this nested+nested structure? something like this:
```go
rv_continue_limit := func(rv, c string, l int64) metav1.ListOptions {
	return metav1.ListOptions{ResourceVersion: rv, Continue: c, Limit: l}
}

switch rv_continue_limit(options.ResourceVersion, options.Continue, options.Limit) {
// initial limited list
case rv_continue_limit("0", "", 4):
	return &v1.PodList{ListMeta: metav1.ListMeta{ResourceVersion: "10"}, Items: pods[0:4]}, nil

// specific version limited list
case rv_continue_limit("10", "", 4):
	return &v1.PodList{ListMeta: metav1.ListMeta{Continue: "C1", ResourceVersion: "11"}, Items: pods[0:4]}, nil

// specific version unlimited list
case rv_continue_limit("10", "", 0):
	return &v1.PodList{ListMeta: metav1.ListMeta{ResourceVersion: "11"}, Items: pods[0:8]}, nil

// Second page of the RV=10 List
case rv_continue_limit("", "C1", 4):
	return nil, apierrs.NewResourceExpired("The resourceVersion for the provided watch is too old.")

default:
	t.Fatalf("unexpected list options: %#v", options)
}
```

Reply:
Neat, done!

Reply:
Also needed to change t.Fatal to panic when realized that t.Fatal doesn't really work as expected inside the List goroutine.